walk-in-interview/src/main/java/com/google/job/servlets/JobsListingsServlet.java

```diff
@@ -0,0 +1,229 @@
+    private static final String REGION_PARAM = "region";
+    private static final String SORT_BY_PARAM = "sortBy";
+    private static final String ORDER_PARAM = "order";
+    private static final String PAGE_SIZE_PARAM = "pageSize";
+    private static final String PAGE_INDEX_PARAM = "pageIndex";
+
+    private JobsDatabase jobsDatabase;
+
+    @Override
+    public void init() {
+        this.jobsDatabase = new JobsDatabase();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        try {
+            
+            JobQuery jobQuery = parseJobQuery(request);
+
+            JobPage jobPage = fetchJobPageDetails(jobQuery);
+
+            String json = ServletUtils.convertToJsonUsingGson(jobPage);
+            response.setContentType("application/json;");
+            response.getWriter().println(json);
+        } catch(IllegalArgumentException | ServletException | ExecutionException | TimeoutException e) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        }
+    }
+
+    /**
+     * Returns the JobPage object.
+     *
+     * @param jobQuery The job query object with all the filtering/sorting params.
+     * @return JobPage object with all the details for the GET response.
+     */
+    private JobPage fetchJobPageDetails(JobQuery jobQuery) throws ServletException, ExecutionException, TimeoutException {
+        try {
+            return this.jobsDatabase.fetchJobPage(jobQuery)
+                    .get(TIMEOUT, TimeUnit.SECONDS);
+        } catch (InterruptedException | IOException e) {
+            throw new ServletException(e);
+        }
+    }
+
+    /**
+     * Returns the job query object.
+     *
+     * @param request From the GET request.
+     * @return the job query object.
+     * @throws IllegalArgumentException if one of the params is invalid.
+     */
+    public static JobQuery parseJobQuery(HttpServletRequest request) throws IllegalArgumentException {
+        int minLimit = parseMinLimit(request);
+        int maxLimit = parseMaxLimit(request);
+        SingaporeRegion region = parseRegion(request);
+        Filter sortBy = parseSortBy(request);
+        Order order = parseOrder(request);
+        // TODO(issue/34): parse page size and index once pagination is implemented
+
+        JobQuery jobQuery = new JobQuery().setMinLimit(minLimit).setMaxLimit(maxLimit).setRegion(region)
+            .setSortBy(sortBy).setOrder(order);
+        
+        return jobQuery;
+    }
+
+    /**
+     * Returns the lower limit as an int.
+     *
+     * @param request From the GET request.
+     * @return the lower limit.
+     * @throws IllegalArgumentException if the page size is invalid.
+     */
+    public static int parseMinLimit(HttpServletRequest request) throws IllegalArgumentException {
+        String minLimitStr = ServletUtils.getStringParameter(request, MIN_LIMIT_PARAM, /* defaultValue= */ "");
```

Comment:
Would the query work without min and max? e.g if not supplied, don't apply those filters? Look like we have some default value set for query builder above?

Reply:
Yep, it would still work without min/max because it would default to the JobQuery value. But I thought it might make sense to have all the filters in the request

Reply:
If the min/max aren't supplied, then they should just become the default values - 0 and Integer.MAX_VALUE